File: front_end/src/model/Address.ts
export type AddressRequest = {
    firstName: string
    lastName: string
    districtId: string
    provinceId: string
    wardId: string
    addressName: string
    phoneNumber?: string
    addressTypeId: string
    customerId?: string
}

export type AddressResponse = {
    id: number
    firstName: string
    lastName: string
    districtId: string
    provinceId: string
    wardId: string
    addressName: string
    phoneNumber: string
    addressTypeId: string
    customerId?: string
}
export type Province = {
    code: string
    nameEn: string
}
export type District = {
    code: string
    nameEn: string
    provinceCode: string
}
export type Ward = {
    code: string
    nameEn: string
    districtCode: string
}

File: src/main/java/com/example/back_end/entity/Address.java
package com.example.back_end.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "address")
public class Address extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @Column(name = "first_name", length = Integer.MAX_VALUE)
    private String firstName;

    @Column(name = "last_name", length = Integer.MAX_VALUE)
    private String lastName;

    @Column(name = "email", length = Integer.MAX_VALUE)
    private String email;

    @Column(name = "company", length = Integer.MAX_VALUE)
    private String company;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "district_id")
    private District district;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "province_id")
    private Province province;

    @ManyToOne(fetch = FetchType.LAZY)
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "ward_id")
    private Ward ward;

    @Column(name = "address_name", length = Integer.MAX_VALUE)
    private String addressName;

    @Column(name = "phone_number", length = 15)
    private String phoneNumber;

}